tiledb/sm/query/global_order_writer.cc

```diff
@@ -779,18 +779,20 @@ Status GlobalOrderWriter::prepare_full_tiles_fixed(
             init_tile_nullable(name, &((*tiles)[i]), &((*tiles)[i + 1])));
 
     // Handle last tile (it must be either full or empty)
```

Comment:
What happens if this is violated? We don't check this condition in a meaningful way below - the assert on line 788 will not be hit in a release build.

Reply:
SC-16777 for follow-up.